### PR TITLE
feat: Add config switch to enable E2EE in browser

### PR DESCRIPTION
**Pull request by artonge**
## Changes

I also added a warning in the mnemonic prompt as E2EE navigation in browser could be maliciously enabled by the admin.

### Settings

Before | After -  no confirmation | After - Bad format | After - All good
-- | -- | -- | --
Screenshot from 2024-12-18 13-50-25 | Screenshot from 2024-12-18 14-42-02 | Screenshot from 2024-12-18 14-41-56 | Screenshot from 2024-12-18 14-41-47

### Mnemonic prompt

View | Before | After
-- | -- | --
All closed | Screenshot from 2024-12-18 13-50-32 | Screenshot from 2024-12-18 13-42-11
Enabled in browser opened and off | NA | Screenshot from 2024-12-18 13-42-17
Enabled in browser opened and on | NA | Screenshot from 2024-12-18 13-42-20
Reset key opened | Screenshot from 2024-12-18 13-50-37 | Screenshot from 2024-12-18 13-42-33

### CODE REVIEW

File: lib/Listener/LoadAdditionalListener.php [hunk 1]
```diff
@@ -10,23 +10,36 @@
...
 			return;
 		}
 
+		$this->initialState->provideInitialState(
+			'userConfig',
+			[
+				'e2eeInBrowserEnabled' => $this->config->getUserValue($this->userId, 'end_to_end_encryption', 'e2eeInBrowserEnabled', 'false') === 'true',
+			]
+		);
+
```
Comment (skjnldsv) [inline lib/Listener/LoadAdditionalListener.php:42]: I would prefer a capability :)

Comment (artonge) [reply lib/Listener/LoadAdditionalListener.php:42]: That a user setting, not a capability of the server, no?

**Review by nfebe — APPROVED** [review]

File: lib/Listener/LoadAdditionalListener.php (hunk 1 above)
Comment (skjnldsv) [reply lib/Listener/LoadAdditionalListener.php:42]: Some capabilities (like theming) changes per user :thinking: 
Maybe it is an antipattern :thinking:

Comment (skjnldsv) [reply lib/Listener/LoadAdditionalListener.php:42]: provisioning_api too https://github.com/nextcloud/server/blob/381077028adf388a7081cf42026570c6be47b198/apps/provisioning_api/lib/Capabilities.php#L32

File: lib/Controller/ConfigController.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			return new JSONResponse([], Http::STATUS_PRECONDITION_FAILED);
+		}
+
+		if (!in_array($key, ['e2eeInBrowserEnabled'])) {
+			return new JSONResponse([], Http::STATUS_PRECONDITION_FAILED);
+		}
+
+		$this->config->setUserValue($this->userId, Application::APP_ID, $key, $value);
+		return new JSONResponse([], Http::STATUS_OK);
+	}
```
Comment (come-nc) [inline lib/Controller/ConfigController.php:43]: Is there no route from core API for this?

**Pull request merged**